reviewer/reviewer.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reviewer
+
+import (
+	"context"
+	"math/rand"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/palantir/policy-bot/policy/common"
+	"github.com/palantir/policy-bot/pull"
+)
+
+func findLeafChildren(result common.Result) []common.Result {
+	var r []common.Result
+	if len(result.Children) == 0 {
+		if result.Status == common.StatusPending && result.Error == nil {
+			return []common.Result{result}
+		}
+	} else {
+		for _, c := range result.Children {
+			if c.Status == common.StatusPending {
+				r = append(r, findLeafChildren(*c)...)
+			}
+		}
+	}
+	return r
+}
+
+// select n random values from the list of users without reuse
+func selectRandomUsers(n int, users []string, r *rand.Rand) []string {
+	selected := make(map[int]bool)
+
+	var selections []string
+	if n == 0 {
+		return selections
+	}
+	if n >= len(users) {
+		return users
+	}
+
+	for i := 0; i < n; i++ {
+		for {
+			i := r.Intn(len(users))
+			if !selected[i] {
+				selected[i] = true
+				selections = append(selections, users[i])
+				break
+			}
+		}
+	}
+	return selections
+}
+
+func shoveIntoMap(m map[string]struct{}, u []string) map[string]struct{} {
+	for _, n := range u {
+		m[n] = struct{}{}
+	}
+	return m
```

Comment:
style: because maps are always references, there's no need to return the input unless it makes for cleaner chaining elsewhere.